src/config.rs

```diff
@@ -54,13 +60,17 @@ impl TryFrom<RawConfig> for Config {
         let auth_during_comm_config =
             AuthDuringCommConfig::try_from(raw_config.auth_during_comm_config)?;
 
+        let oauth_provider = auth::OauthProvider::from_str(&raw_config.oauth_provider)?;
```

Comment:
Since we control the OauthProvider type we should implement Deserializer directly for OauthProvider, which will make migrating away from rawconfig easier once try_from for fields lands.

Reply:
We will change this in a new PR